src/docfx/lib/json/SourceInfoJsonConverter.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Newtonsoft.Json;
+
+namespace Microsoft.Docs.Build
+{
+    internal class SourceInfoJsonConverter : JsonConverter
+    {
+        public override bool CanConvert(Type objectType)
+        {
+            return objectType.IsGenericType && objectType.GetGenericTypeDefinition() == typeof(SourceInfo<>);
+        }
+
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            var range = JsonUtility.ToRange((IJsonLineInfo)reader);
+            var valueType = objectType.GenericTypeArguments[0];
+            var value = serializer.Deserialize(reader, valueType);
+
+            // TODO: populate file info
```

Comment:
just out of curiosity, is it possible to get the file info here?

Reply:
through some thread static foo